lib/eth/abi/decoder.rb

```diff
@@ -48,7 +61,7 @@ def type(type, arg)
           # decoded dynamic-sized arrays
           (0...l).map { |i| type(nested_sub, arg[32 + nested_sub.size * i, nested_sub.size]) }
         elsif !type.dimensions.empty?
-          l = type.dimensions.last[0]
+          l = type.dimensions.first
```

Comment:
To be honest, I'm not sure what I did here 😅 but now all the tests pass, and it fixes the problem with static-size array decoding. I tried to check in git history why this was introduced but could not find anything understandable for me. Please share your feedback

Reply:
You would be surprised how old this line is.

https://github.com/cryptape/ruby-ethereum-abi/blob/90d4fa3fc6b568581165eaacdc506b9b9b49e520/lib/ethereum/abi.rb#L243